components/eam/src/physics/cam/zm_conv_intr.F90

```diff
@@ -1345,378 +1197,329 @@ subroutine zm_conv_tend(pblh    ,mcon    ,cme     , &
+   call physics_ptend_init(ptend, state%psetcols, 'convtran2', lq=lq )
 
-  lchnk = state%lchnk
+   ! Associate pointers with physics buffer fields
+   ifld = pbuf_get_index('FRACIS')
+   call pbuf_get_field(pbuf, fracis_idx, fracis, start=(/1,1,1/), kount=(/pcols, pver, pcnst/) )
 
-   nstep = get_nstep()
-   if((convproc_do_aer .or. convproc_do_gas) .and. clim_modal_aero) then
+   ! Transport all constituents except cloud water and ice
+   if ( (convproc_do_aer .or. convproc_do_gas) .and. clim_modal_aero ) then
       do m = 1, pcnst
-         if ( (species_class(m) == spec_class_aerosol .and. convproc_do_aer) .or. &
-              (species_class(m) == spec_class_gas     .and. convproc_do_gas) ) then
+         if ( (species_class(m)==spec_class_aerosol .and. convproc_do_aer) .or. &
+              (species_class(m)==spec_class_gas     .and. convproc_do_gas) ) then
             ptend%lq(m) = .false.
          end if
-      enddo
-   endif
-
+      end do
+   end if
 
    if (any(ptend%lq(:))) then
-      ! initialize dpdry for call to convtran
-      ! it is used for tracers of dry mixing ratio type
-      dpdry = 0._r8
+      ! initialize dpdry for call to convtran for tracers of dry mixing ratio type
+      dpdry(1:ncol,1:pver) = 0
       do i = 1,lengath
-         dpdry(i,:) = state%pdeldry(ideep(i),:)/100._r8
+         dpdry(i,1:pver) = state%pdeldry(ideep(i),1:pver)/100
       end do
-
       call t_startf ('convtran2')
-      call convtran (lchnk,                                        &
-                     ptend%lq,state%q, pcnst,  mu, md,   &
-                     du, eu, ed, dp, dsubcld,  &
-                     jt,maxg,ideep, 1, lengath,  &
-                     nstep,   fracis,  ptend%q, dpdry,  ztodt)  
+      call convtran (lchnk, ptend%lq, state%q, pcnst,  mu, md, &
+                     du, eu, ed, dp, dsubcld,  jt, maxg, ideep, 1, lengath, &
+                     nstep, fracis, ptend%q, dpdry, ztodt)
       call t_stopf ('convtran2')
    end if
 
-   if((convproc_do_aer .or. convproc_do_gas) .and. clim_modal_aero) then
+   if ( (convproc_do_aer .or. convproc_do_gas) .and. clim_modal_aero ) then
       do m = 1, pcnst
-         if ( (species_class(m) == spec_class_aerosol .and. convproc_do_aer) .or. &
-              (species_class(m) == spec_class_gas     .and. convproc_do_gas) ) then
+         if ( (species_class(m)==spec_class_aerosol .and. convproc_do_aer) .or. &
+              (species_class(m)==spec_class_gas     .and. convproc_do_gas) ) then
             ptend%lq(m) = .false.
-            ptend%q(:,:,m) = 0.0_r8
+            ptend%q(1:ncol,1:pver,m) = 0
          end if
-      enddo
-   endif
+      end do
+   end if
 
 end subroutine zm_conv_tend_2
 
+!===================================================================================================
 
 subroutine zm_conv_micro_outfld(microp_st, dlf, dif, dnlf, dnif, frz, lchnk, ncol)
+   !----------------------------------------------------------------------------
```

Comment:
`use cam_history,   only: outfld` is removed in this subroutine but not in others. Is there a reason for this?

Reply:
ah right, I meant to remove the other and have a single use statement at the top of the module for things like this. It's fixed in the latest commits.